ocflib/lab/hours2.py

```diff
@@ -0,0 +1,289 @@
+
+All times are assumed to be OST (OCF Standard Time).
+
+Usage:
+
+    >>> from ocflib.lab.hours2 import read_hours_listing()
+    >>> hours_listing = read_hours_listing()
+    >>> hours_listing.hours_on_date(date(2015-10-12))
+    [Hour(open=datetime.time(9, 0), close=datetime.time(18, 0))]
+"""
+from datetime import date
+from datetime import datetime
+from datetime import time
+from datetime import timedelta
+from enum import IntEnum
+
+import attr
+import yaml
+
+
+class Weekday(IntEnum):
+    Monday = 0
+    Tuesday = 1
+    Wednesday = 2
+    Thursday = 3
+    Friday = 4
+    Saturday = 5
+    Sunday = 6
+
+
+def _parsetime(t):
+    if isinstance(t, time):
+        return t
+
+    return datetime.strptime(t, '%H:%M').time()
+
+
+def _parse_regular_hours(regular):
+    """Parses a dictionary of regular hours, as used in the yaml file.
+
+    Keys can be either weekday strings ("Monday", "Tuesday", ...) or Weekday
+    enums. Values should be a list of timeranges, parseable by
+    _parse_hours_list.
+    """
+    out = {}
+    for weekday, hours in regular.items():
+        try:
+            # First see if it's an int already
+            weekday = Weekday(weekday)
+        except ValueError:
+            # Convert strings ('Monday') to int (0)
+            weekday = getattr(Weekday, weekday)
+
+        out[weekday] = _parse_hours_list(hours)
+
+    if out.keys() != set(Weekday):
+        raise ValueError('Regular hours must be set for all weekdays')
+
+    return out
+
+
+def _parse_hours_list(time_ranges):
+    """Converts a list of timeranges to a list of Hour objects.
+
+    This function idempotently does nothing on a list of Hour objects.
+
+    Raises an ValueError if the timeranges overlap.
+
+    >>> _parse_hours_list([['8:30', '10:00'], ['11:00', '3:00']])
+    [Hour(time(8, 30), time(10)), Hour(time(11), time(3))]
+
+    >>> _parse_hours_list([['8:30', '10:00'], ['9:00', '11:00']])
+    Traceback (most recent call last):
+        ...
+    ValueError: Hours must be in order
+
+    >>> _parse_hours_list([['8:30', '10:00'], ['8:00', '11:00']])
+    Traceback (most recent call last):
+        ...
+    ValueError: Hours must be in order
+    """
+    hours = []
+
+    for hour in time_ranges:
+        if not isinstance(hour, Hour):
+            hour = Hour(hour[0], hour[1])
+
+        if hours and hours[-1].close > hour.open:
+            raise ValueError('Hours must be in order')
+
+        hours.append(hour)
+
+    return hours
+
+
+def _parse_holiday(holiday):
+    """Converts a holiday dict to a Holiday object.
+
+    This function idempotently does nothing on a Holiday object.
+    """
+    if isinstance(holiday, Holiday):
+        return holiday
+
+    if isinstance(holiday['date'], list):
+        start, end = holiday['date']
+    elif isinstance(holiday['date'], date):
+        start = end = holiday['date']
+    else:
+        raise ValueError(
+            'Holiday date is not a (start, end) pair or single datetime'
```

Comment:
nit: if we expect a list and not a tuple, you should say `is not a [start, end] pair`
```suggestion
            'Holiday date is not a [start, end] pair or single datetime'
```